### PR TITLE
Login/Create Account - REWORK

### PR DESCRIPTION
**Pull request by OvidiuCalin123**
Rework covering these stories: 

[Issue 29](https://dev.azure.com/NessLabs/TimisoaraEvents/_workitems/edit/29): Icons
[Issue 12](https://dev.azure.com/NessLabs/TimisoaraEvents/_workitems/edit/12): Login page for admin - UI
[Issue 20](https://dev.azure.com/NessLabs/TimisoaraEvents/_workitems/edit/20): Create account page for admin - UI

### CODE REVIEW

**Comment by OvidiuCalin123** [comment]
Worked with Rossana on this PR

File: ness-lab-angular/src/app/components/create-account/create-account.component.html [hunk 1]
```diff
@@ -0,0 +1,184 @@
+<div
+  class="container"
+  style="
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/create-account/create-account.component.html:3]: move this to a scss class

File: ness-lab-angular/src/app/components/create-account/create-account.component.scss [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+.password-requirement-title {
+  font-size: 55%;
+  font-family: Arial, Helvetica, sans-serif;
+  font-weight: bold;
+  padding-left: 10%;
+  padding-bottom: 3px;
+  margin-top: 15px;
+  transform: scale(0.9);
+}
+* {
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/create-account/create-account.component.scss:30]: you don't need this, is already applied to html tag in index html

File: ness-lab-angular/src/app/components/create-account/create-account.component.ts [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+      this.passwordRequirementBulletCheck[3][1]
+    ) {
+      this.conditionPasswordRegexes = true;
+    } else {
+      this.conditionPasswordRegexes = false;
+      this.buttonEnabled = false;
+    }
+
+    if (
+      passwordValue !== '' &&
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/create-account/create-account.component.ts:105]: you can use the form to validate the button

File: ness-lab-angular/src/app/components/create-account/create-account.component.ts [hunk 4]
```diff
@@ -0,0 +1,166 @@
...
+    const emailValue = emailControl.value;
+
+    const usernameControl = this.createAccountForm.get('username');
+    const usernameValue = usernameControl.value;
+
+    const passwordControl = this.createAccountForm.get('password');
+    const passwordValue = passwordControl.value;
+
+    if (
+      this.passwordRequirementBulletCheck[0][1] &&
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/create-account/create-account.component.ts:93]: don't need any extra checks, just use the form to see if it's valid or not

File: ness-lab-angular/src/app/components/login/login.component.html [hunk 5]
```diff
@@ -0,0 +1,133 @@
+<div
+  class="container"
+  style="
+    -webkit-background-size: cover;
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.html:4]: move in a scss class

File: ness-lab-angular/src/app/components/login/login.component.html [hunk 6]
```diff
@@ -0,0 +1,133 @@
...
+        <mat-error
+          class="invalid-feedback"
+          *ngIf="loginForm.get('email').hasError('email')"
+        >
+          The email is not in the correct format.
+        </mat-error>
+      </div>
+      <div class="form-group was-validated">
+        <label class="form-label" for="password">
+          <svg
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.html]: why do you use this?

File: ness-lab-angular/src/app/components/login/login.component.scss [hunk 7]
```diff
@@ -0,0 +1,136 @@
+* {
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.scss:1]: remove this

File: ness-lab-angular/src/app/components/login/login.component.scss [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+  margin: 0;
+  padding: 0;
+  box-sizing: border-box;
+
+  font-family: Arial, Helvetica, sans-serif;
+}
+
+body {
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.scss]: remove this

File: ness-lab-angular/src/app/components/login/login.component.ts [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+
+  setButtonEnabled($event) {
+    this.buttonEnabled = $event.buttonEnabled;
+  }
+
+  onSubmit() {
+    console.log('OK');
+  }
+
+  canSetButtonEnabled() {
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.ts:45]: use form property to see if it's valid or not

File: ness-lab-angular/src/app/generic/form/form.component.html [hunk 10]
```diff
@@ -0,0 +1,9 @@
+<form [formGroup]="form" (ngSubmit)="onSubmit()">
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/generic/form/form.component.html:1]: remove app-input and replace it with material

File: ness-lab-angular/src/app/generic/form/form.component.ts [hunk 11]
```diff
@@ -0,0 +1,119 @@
+import { Component, Input, OnInit, Output, EventEmitter } from '@angular/core';
+import { FormControl, FormGroup, Validators } from '@angular/forms';
+import { FormInputBase } from '../../model/form-input-base.model';
+
+interface eventsEmitted {
+  passwordRequirementBulletCheck: {};
+  buttonEnabled: boolean;
+}
+
+@Component({
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/generic/form/form.component.ts:10]: remove this component if you don't use it

File: ness-lab-angular/src/app/model/form-input-base.model.ts [hunk 12]
```diff
@@ -0,0 +1,46 @@
+import { ValidatorFn } from '@angular/forms';
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/model/form-input-base.model.ts:1]: move this in a interface

File: ness-lab-angular/src/app/model/form-dropdown.model.ts [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import { FormInputBase } from './form-input-base.model';
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/model/form-dropdown.model.ts:1]: you don't need this anymore

File: ness-lab-angular/src/app/model/form-checkbox.model.ts [hunk 14]
```diff
@@ -0,0 +1,5 @@
+import { FormInputBase } from './form-input-base.model';
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/model/form-checkbox.model.ts:1]: remove this

File: ness-lab-angular/src/app/model/form-textbox.model.ts [hunk 15]
```diff
@@ -0,0 +1,5 @@
+import { FormInputBase } from './form-input-base.model';
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/model/form-textbox.model.ts:1]: remove this

File: ness-lab-angular/src/app/components/create-account/create-account.component.html [hunk 16]
```diff
@@ -0,0 +1,184 @@
...
+    <form class="needs-validation" [formGroup]="createAccountForm">
+      <img
+        src="../assets/img/logo-timisoara-2023.png"
+        class="img-capitala-europeana"
+        alt="capitala-europeana"
+      />
+
+      <p class="text">Create Account</p>
+
+      <div class="form-group was-validated">
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/create-account/create-account.component.html]: use material component in here

File: ness-lab-angular/src/app/components/login/login.component.html [hunk 17]
```diff
@@ -0,0 +1,133 @@
...
+        src="../assets/img/logo-timisoara-2023.png"
+        class="img-capitala-europeana"
+        alt="capitala-europeana"
+      />
+
+      <p class="text">Login</p>
+
+      <div class="form-group was-validated">
+        <label class="form-label" for="email">
+          <svg
```
Comment (MilosRafael) [inline ness-lab-angular/src/app/components/login/login.component.html]: use mat icon here instead of svg

**Pull request merged**